cmd/client.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Author: Evan Spendlove (@evanSpendlove)
+//
+// Client.go creates the client for connecting to, and making requests to, the
+// gRPC server of Cloudprober.
+//
+// TODOs:
+// TODO(#38) Add support for secure connection to RPC server with credentials, if supported by Cloudprober.
+// TODO: Include details of these errors (addProbeFromConfig method) in the error message.
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/golang/glog"
+	proberpb "github.com/google/cloudprober/prober/proto"
+	cpprobes "github.com/google/cloudprober/probes"
+	configpb "github.com/google/cloudprober/probes/proto"
+	"google.golang.org/grpc"
+)
+
+type CloudproberClient struct {
+	conn      *grpc.ClientConn           // Client connection for opening and closing the connection
+	client    proberpb.CloudproberClient // Cloudprober gRPC client retained between RPC calls
+	clientMux sync.Mutex                 // Mutex lock for making CloudproberClient safe for concurrent use
```

Comment:
The convention in Go is to declare the mutex just above the fields it's meant to protect.

Reply:
I have amended this so that the mutex is above the `client` field.